_Bool test

```c
#include<stdio.h>
_Bool f(_Bool b){
        return  b;
}
int main(){
        _Bool b = 0;
        if (b){
                b++;
        }
        _Bool c = f(b);
        b = b + c;
        int i = (int)(b);
        printf("%d\n",i);
        return 0;
}
```
```
○ → gcc 2.c; ./a.out
0
```
Go:
```go
func f(b bool) bool {
        return b
}
func main() {
        __init()
        var b bool = 0 != 0
        if b {
                func() bool {
                        b += 1
                        return b
                }()
        }
        var c bool = f(b)
        b = b+c
        var i int = map[bool]int{false: 0, true: 1}[map[bool]int{false: 0, true: 1}[(b)]]
        noarch.Printf([]byte("%d\n\x00"), i)
        return
}
```
```
○ → go build 2.go
# command-line-arguments
./2.go:190:6: cannot convert 1 to type bool
./2.go:190:6: invalid operation: b += 1 (mismatched types bool and int)
./2.go:195:7: invalid operation: b + c (operator + not defined on bool)
./2.go:196:45: cannot use map[bool]int literal[b] (type int) as type bool in map index
```

--- a/tests/stdbool.c
+++ b/tests/stdbool.c
@@ -2,9 +2,13 @@
 #include <stdbool.h>
 #include "tests.h"
 
+_Bool f(_Bool b){
+        return  b;
+}
+
 int main()
 {
-    plan(9);
+    plan(12);
 
     bool trueBool = true;
     bool falseBool = false;
@@ -76,6 +80,26 @@ int main()
 	else
 	{
         fail("%s", "should not reach here")
+	}
+
+
+	_Bool b = 0; // false
+	if (b){ b++;}
+	if (b == false) // b = 0
+	{
+        pass("%s", "ok")
+	}
+
+	_Bool c = f(b);
+	b = b + c;
+	if (b == false)
+	{
+        pass("%s", "ok")
+	}
+	int i = (int)(b);
+	if (i == 0)
+	{
+        pass("%s", "ok")
 	}
 
     done_testing();

--- a/tests/stdbool.c
+++ b/tests/stdbool.c
@@ -4,7 +4,7 @@
 
 int main()
 {
-    plan(6);
+    plan(9);
 
     bool trueBool = true;
     bool falseBool = false;
@@ -47,6 +47,36 @@ int main()
     {
         fail("%s", "should not reach here")
     }
+
+	_Bool var = true;
+	if(var)
+	{
+        pass("%s", "ok")
+	}
+    else
+    {
+        fail("%s", "should not reach here")
+    }
+
+	var = true;
+	if(var-var)
+	{
+        fail("%s", "should not reach here")
+	}
+	else
+	{
+        pass("%s", "ok")
+	}
+
+	var = true;
+	if(var - var == false)
+	{
+        pass("%s", "ok")
+	}
+	else
+	{
+        fail("%s", "should not reach here")
+	}
 
     done_testing();
 }

--- a/transpiler/unary.go
+++ b/transpiler/unary.go
@@ -94,7 +94,7 @@ func transpileUnaryOperatorNot(n *ast.UnaryOperator, p *program.Program) (
 		eType = "int"
 	}
 
-	if eType == "bool" || eType == "_Bool" {
+	if eType == "bool" {
 		return &goast.UnaryExpr{
 			X:  e,
 			Op: token.NOT,

--- a/transpiler/unary.go
+++ b/transpiler/unary.go
@@ -94,13 +94,6 @@ func transpileUnaryOperatorNot(n *ast.UnaryOperator, p *program.Program) (
 		eType = "int"
 	}
 
-	if eType == "bool" || eType == "_Bool" {
-		return &goast.UnaryExpr{
-			X:  e,
-			Op: token.NOT,
-		}, "bool", preStmts, postStmts, nil
-	}
-
 	if strings.HasSuffix(eType, "*") {
 		// `!pointer` has to be converted to `pointer == nil`
 		return &goast.BinaryExpr{

--- a/transpiler/unary.go
+++ b/transpiler/unary.go
@@ -94,6 +94,13 @@ func transpileUnaryOperatorNot(n *ast.UnaryOperator, p *program.Program) (
 		eType = "int"
 	}
 
+	if eType == "bool" || eType == "_Bool" {
+		return &goast.UnaryExpr{
+			X:  e,
+			Op: token.NOT,
+		}, "bool", preStmts, postStmts, nil
+	}
+
 	if strings.HasSuffix(eType, "*") {
 		// `!pointer` has to be converted to `pointer == nil`
 		return &goast.BinaryExpr{

--- a/types/cast.go
+++ b/types/cast.go
@@ -67,6 +67,14 @@ func CastExpr(p *program.Program, expr goast.Expr, cFromType, cToType string) (g
 	fromType := cFromType
 	toType := cToType
 
+	// registration type _Bool in program
+	if cFromType == "_Bool" {
+		p.TypedefType["_Bool"] = "int"
+	}
+	if cToType == "_Bool" {
+		p.TypedefType["_Bool"] = "int"
+	}
+
 	// Checking registated typedef types in program
 	if v, ok := p.TypedefType[toType]; ok {
 		if fromType == v {
@@ -334,7 +342,7 @@ func CastExpr(p *program.Program, expr goast.Expr, cFromType, cToType string) (g
 		return util.NewStringLit(`""`), nil
 	}
 
-	if fromType == "_Bool" && toType == "bool" {
+	if fromType == "_Bool" && toType == "int" {
 		return expr, nil
 	}
 

--- a/types/cast_test.go
+++ b/types/cast_test.go
@@ -41,11 +41,6 @@ func TestCast(t *testing.T) {
 
 		// Casting to bool
 		{args{util.NewIntLit(1), "int", "bool"}, util.NewBinaryExpr(util.NewIntLit(1), token.NEQ, util.NewIntLit(0), "bool", false)},
-
-		// String types
-		// {args{"foo", "[3]char", "const char*"}, "1 != 0"},
-
-		// {args{util.NewIdent("false"), "_Bool", "int"}, util.NewIdent("false")},
 	}
 
 	for _, tt := range tests {

--- a/types/cast_test.go
+++ b/types/cast_test.go
@@ -45,7 +45,7 @@ func TestCast(t *testing.T) {
 		// String types
 		// {args{"foo", "[3]char", "const char*"}, "1 != 0"},
 
-		{args{util.NewIdent("false"), "_Bool", "bool"}, util.NewIdent("false")},
+		// {args{util.NewIdent("false"), "_Bool", "int"}, util.NewIdent("false")},
 	}
 
 	for _, tt := range tests {

--- a/types/resolve.go
+++ b/types/resolve.go
@@ -39,7 +39,7 @@ var simpleResolveTypes = map[string]string{
 	"unsigned short":         "uint16",
 	"unsigned short int":     "uint16",
 	"void":                   "",
-	"_Bool":                  "bool",
+	"_Bool":                  "int",
 
 	// void*
 	"void*":  "interface{}",
@@ -116,6 +116,10 @@ var NullPointer = "NullPointerType *"
 //    until a more suitable solution is found for those cases.
 func ResolveType(p *program.Program, s string) (string, error) {
 	s = CleanCType(s)
+
+	if s == "_Bool" {
+		p.TypedefType[s] = "int"
+	}
 
 	// FIXME: This is a hack to avoid casting in some situations.
 	if s == "" {

--- a/types/resolve_test.go
+++ b/types/resolve_test.go
@@ -19,7 +19,7 @@ var resolveTestCases = []resolveTestCase{
 	{"__uint16_t", "uint16"},
 	{"void *", "interface{}"},
 	{"unsigned short int", "uint16"},
-	{"_Bool", "bool"},
+	{"_Bool", "int"},
 	{"struct RowSetEntry *", "[]RowSetEntry"},
 	{"div_t", "noarch.DivT"},
 	{"ldiv_t", "noarch.LdivT"},

